### PR TITLE
Allow 'mmctl channel move' to move private channels

### PR DESCRIPTION
**Pull request by Leats**
#### Summary

Currently `mmctl channel move someotherteam teamname:channelname` does not allow moving private channels. The error message only mentions direct message and group message channels that cannot be moved:
https://github.com/mattermost/mattermost-server/blob/15037a510a5da29bf16badb5ae4e6c69ee6fe946/i18n/en.json#L339-L340
While not being able to move direct message and group message channels makes sense, the same should not apply to private channels.

#### Ticket Link

None

#### Release Note

```release-note
Fixed an issue where `mmctl channel move` did not allow moving private channels.
```

### CODE REVIEW

**Review by ThiefMaster — APPROVED** [review]
Very useful fix! I had to make a channel temporarily public. move it, and then make it private again as a workaround (ugly, because in theory this could leak data if someone managed to join just in that moment), and the old `mattermost channel move ...` command did not have this problem either.

**Review by streamer45 — APPROVED** [review]
Thank you @Leats.

Changes look good to me. I am just not sure if this was an overlook or if there was a reason to limit this to public channels only. Will defer to @mgdelacroix on that.

**Comment by jfrerich** [comment]
@mgdelacroix, ping for review (and welcome back from the holiday break)

**Review by mgdelacroix — COMMENTED** [review]
Thanks a lot for the contribution, @Leats! The change LGTM, I would only like to increase the test coverage.

Would you be open to add test cases for `Should fail to move channel due to a member not member of target team` and `Should be able to (force) move channel by a member that is not member of target team` with a private channel? Both exist already in the same test function for public channels.

**Comment by Leats** [comment]
> Would you be open to add test cases for `Should fail to move channel due to a member not member of target team` and `Should be able to (force) move channel by a member that is not member of target team` with a private channel?

Thank you! Yes, I can add that as well! ☺️

**Comment by emilyacook** [comment]
> This PR has been automatically labelled "stale" because it hasn't had recent activity.
> A core team member will check in on the status of the PR to help with questions.
> Thank you for your contribution!
> 

Hey @Leats! Let me know if there is any update on this :)

**Comment by Leats** [comment]
Hello @emilyacook @mgdelacroix! 
Very sorry for the wait, my move took longer than expected. I added the tests as requested, I hope everything is alright! :)

File: api4/channel_test.go
Comment (mgdelacroix) [inline api4/channel_test.go:4031]:
```suggestion
		_, resp = client.MoveChannel(privateChannel.Id, team2.Id, false)
```

File: api4/channel_test.go
Comment (mgdelacroix) [inline api4/channel_test.go:4061]:
```suggestion
		newChannel, resp := client.MoveChannel(privateChannel.Id, team2.Id, true)
```

**Review by mgdelacroix — CHANGES_REQUESTED** [review]
Thanks for adding the tests @Leats! Added a couple of small changes

**Comment by Leats** [comment]
Thank you, @mgdelacroix! I changed the code as requested and added the same behaviour to the local channel move so it's consistent with my previous changes (and lets the tests run through).

**Review by mgdelacroix — APPROVED** [review]
Thanks a lot @Leats! LGTM!! :+1:

**Comment by lindy65** [comment]
Hi @mgdelacroix - Can you link me to the Jira ticket for this PR please? Thanks :)

**Comment by ThiefMaster** [comment]
I don't think there is one for it - it's just an annoying issue we discovered on our Mattermost instance when moving channels for a user, so I suggested @Leats that sending a PR for this would be a nice contribution.

**Comment by lindy65** [comment]
Thanks @ThiefMaster - QA requires a Jira ticket to move the PR through the process so wondering whether you'd like to create one? :)

**Comment by ThiefMaster** [comment]
I don't think people who aren't mattermost staff can do that, can they?

> You do not have permission to create issues in this project.

**Comment by lindy65** [comment]
Ah, you're correct @ThiefMaster - non-staff can view Jira tickets only.  So @mgdelacroix will need to help us out here :)

**Comment by lindy65** [comment]
Hi @mgdelacroix :) I've created a [jira ticket](https://mattermost.atlassian.net/browse/MM-32746) for this PR. Please help fill in QA test steps and description?  TIA 👍

**Comment by mgdelacroix** [comment]
Thanks @lindy65!! done :)

**Review by lindy65 — APPROVED** [review]
Thanks @Leats - looks good to me.  Testing will be done after merge 👍

**Pull request merged**
